src/factor_graph.cc

```diff
@@ -147,16 +202,53 @@ void FactorGraph::safety_check() {
   }
 }
 
+template <class T>
+void parallel_copy(const std::unique_ptr<T[]> &src, std::unique_ptr<T[]> &dst,
+                   size_t num) {
```

Comment:
We could put this in the header where `COPY_ARRAY_UNIQUE_PTR_MEMBER` is, so others can reuse?